plugin/completion.py

```diff
@@ -200,9 +178,11 @@ def handle_response(self, response: Optional[Union[dict, List]],
 
         if response_incomplete:
             flags |= sublime.DYNAMIC_COMPLETIONS
-
         completion_list.set_completions(items, flags)
-        CompletionHandler.last_prefix = prefix
+
+        for point in locations:
+            # save the line to restore later (only when selecting a completion item with a TextEdit)
+            RestoreLines.save_line(point, self.view)
```

Comment:
You'd want to do this in `on_query_completions`, as (assuming `client.send_request` is non-blocking) the user can type before the response comes back

Reply:
For some reason I thought on_query_completions was too early. 
But you are right.

Reply:
Moved it
Thanks :)